### ISSUE TITLE
Add full implementation of paintgun and related entities

### ISSUE THREAD

**Issue by SpyceTewan** [ChaosInitiative/Portal-2-Community-Edition#248]
This issue includes the paintgun functionality, model, texturing and interaction with fizzlers and refill cans

**Comment by FrozenDragon0** [comment]
Will there be an option to make the paintgun not require refill cans for modders and just use the activation fizzlers?

**Comment by SpyceTewan** [comment]
Yes the plan is that cans refill the internal tank directly but are limited in quantity, so you can only shoot a limited amount of gel until it's empty. Activation fields (similar to aperture tag but not entirely the same in terms of appearance) can be used to activate a gel color, which can then be shot indefinitely until it's deactivated.

Normal fizzlers always fizzle the paint ammo

**Comment by FrozenDragon0** [comment]
Alright cool.

**Comment by Luke18033** [comment]
Can there also be VScript functions to activate/deactivate different gel types?
